test/unit/search/fm_index/fm_index_test_template.hpp

```diff
@@ -100,9 +101,10 @@ TYPED_TEST_P(fm_index_test, serialization)
 
 TYPED_TEST_P(fm_index_test, concept_check)
 {
-    EXPECT_TRUE(FmIndex<TypeParam>);
-    EXPECT_TRUE(FmIndexTraits<fm_index_default_traits>);
-    EXPECT_TRUE(BiFmIndexTraits<bi_fm_index_default_traits>);
+    if constexpr (detail::is_type_specialisation_of_v<TypeParam, fm_index>)
+        EXPECT_TRUE(FmIndex<TypeParam>);
+    else
+        EXPECT_TRUE(BiFmIndex<TypeParam>);
```

Comment:
Does a `bi_fm_index` also satisfy the `FmIndex` concept? (which I would expect)
If yes, then both if/else branches should be executed for this type.

Reply:
`bi_fm_index` also satisfies the `FmIndex`, but the `BiFmIndex` is a refinement of `FmIndex`, i.e. it requires that the index is an `FmIndex` and then some more stuff.
So I think it is fine this way?

Reply:
yes, okay!

Reply:
So why not just testing:
```cpp    
EXPECT_TRUE(FmIndex<TypeParam>);
if constexpr (detail::is_type_specialisation_of_v<TypeParam, bi_fm_index>)
    EXPECT_TRUE(BiFmIndex<TypeParam>);
```
?